### ISSUE TITLE
LC0051: Missing warning when assigning a label to a field of type Text with defined length

### ISSUE THREAD

**Issue by DennisFranzmannGOB** [StefanMaron/BusinessCentral.LinterCop#936]
Hello,

While refactoring code for the recently added LC0051 ruleset, I noticed the following:

Given: I have a Label Variable without a defined MaxLength and a TableField of type Text/Code with a defined Length X.
When: I assign that label to the field, it does not trigger an LC0051 warning.

Is this known, a potential bug or a potential enhancement?

AL Language Version: 14.2.1249978
Recent LinterCop Version

Example: 

```
procedure AssignText()
var
    Item: Record Item;
    TextVariable: Text[10];
    ExampleTextLbl: Label 'Example Text';
begin
    TextVariable := ExampleTextLbl; //Throws a warning AA0139: Possible overflow assigning 'Label' to 'Text[10]'
    Item.Validate(Description, ExampleTextLbl); //Throws a warning LC0051: Possible overflow assigning 'Label' to 'Text[100]'
    Item.Description := ExampleTextLbl; //Throws no warning
end;
```

**Comment by Arthurvdv** [comment]
@DennisFranzmannGOB Thank you for raising this issue 🙋‍♂️

I had some time this week and did some more research on the [CodeCop AA0139](https://learn.microsoft.com/en-us/dynamics365/business-central/dev-itpro/developer/analyzers/codecop-aa0139) diagnostic based on your example.

```AL
procedure AssignText()
var
    Item: Record Item;
    TextVariable: Text[10];
    ExampleTextLbl: Label 'Example Text';
begin
    TextVariable := ExampleTextLbl;             // AA0139
    Item.Validate(Description, ExampleTextLbl); // LC0051
    Item.Description := ExampleTextLbl;         // Nothing here ..
end;
```

The AA0139 will raise a diagnostic if the text of the Label variable is already longer then the target variable you want to assign it to.

```AL
procedure AssignText()
var
    TextVariable: Text[20];
    TextVariableSmall: Text[10];
    ExampleTextLbl: Label 'Example Text';
begin
    TextVariable := ExampleTextLbl;      // Does not raises AA0139, because 12 => 20
    TextVariableSmall := ExampleTextLbl; // Raises AA0139, because 12 => 10
end;
```

This is the reason why the `Item.Description := ExampleTextLbl` in your example doesn't raise a diagnostic. I believe this could potentially lead to an runtime error, when an translation is created for the `ExampleTextLbl` variable, which could results in a longer text.

In the Pre-release [v0.39.1](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.39.1) version of the LinterCop I've implemented the change for LC0051 to handle this specific scenario for Label type variables.

**Comment by Arthurvdv** [comment]
This is now resolved in the [v0.40.0](https://github.com/StefanMaron/BusinessCentral.LinterCop/releases/tag/v0.40.0) version of the LinterCop.
